content/intro-to-onchain.md

```diff
@@ -0,0 +1,96 @@
+
+# Overview
+
+Each Solana cluster (mainnet-beta, testnet, devnet, localnet) is effectively a single computer with globally syncronized state. The programs that run on Solana - the ones that create tokens, swap tokens, art marketplaces, escrows, market makers, DePIN apps, auctions, retail payments platforms, etc - are called **Solana apps**.
+
+The most popular way to build onchain apps is using **Rust** language and the **Anchor** framework. Frameworks for other languages, including **TypeScript**, **Python** and **Move** are coming soon. 
+
+These frameworks implement common security checks automatically, and handle common tasks like:
+ - sending incoming instructions to the right instruction handlers
+ - deserialising data from incoming transactions
+ - checking the accounts provided with incoming instructions, for example to check that certain accounts are of a particular type, or are distinct from other accounts.
+
+Regardless of the language and framework you choose, Solana works the same. Let's refresh how programs work on Solana.
+
+![Diagram showing a transaction with twp instructions](../assets/transaction-and-instructions.svg)
+
+## Programs are deployed at addresses
+
+Like like we can send tokens to users using their public key, we can find programs using the program's public key. When using Anchor, a keypair is created during `anchor init` and the private key saved in the `target/deploy` directory of your project.
+
+A program's public key is sometimes called a 'program ID'.
+
+## Programs have instruction handlers
+
+For example, a Solana client making a transaction transferring some USDC with a memo saying 'thanks' would have two instructions:
+  - one instruction for the Token program's
+`transfer` instruction handler
+  - the other instruction for the Memo program's `memo` instruction handler. 
+
+Both these instructions must complete successfully for the transaction to execute.
+
+Instruction handlers are how blockchain programs process the instructions from clients. Every exchange, lending protocol, escrow, oracle, etc. provides their functionality by instruction handlers. 
+
+## Instruction handlers write their state to Solana accounts
+
+If you have previously done web development, you can think of instruction handlers like an HTTP route handler, and incoming instructions like HTTP requests. 
+
+But unlike HTTP route handlers, Solana instruction handlers don't return data. Instead, the instruction handlers write their data to accounts on Solana.
+
+Programs on Solana can transfer tokens, which end up in user wallet addresses (for SOL) for the users token accounts (for other tokens). 
+
+But more importantly, programs on Solana can create additional addresses as needed, to store items of data. 
+
+## Programs store data in Program Derived Addresses (PDAs), a key-value store
+
+These addresses are called **program derived addresses** and act as a **key / value store**.
```

Comment:
For the sake of scanability, I would prefer that we start each section with something that doesn't depend on the last sentence of the previous sentence, so instead of "these addresses are called...." I would reword to "Addresses created by Solana programs are called program derived addresses..."